ssd1306/ssd1306_spi.go

```diff
@@ -0,0 +1,68 @@
+package ssd1306
+
+import (
+	"machine"
+	"time"
+
+	"tinygo.org/x/drivers"
+)
+
+type SPIBus struct {
+	wire     drivers.SPI
+	dcPin    machine.Pin
+	resetPin machine.Pin
+	csPin    machine.Pin
+	buffer   []byte // buffer to avoid heap allocations
+}
+
+// NewSPI creates a new SSD1306 connection. The SPI wire must already be configured.
+func NewSPI(bus drivers.SPI, dcPin, resetPin, csPin machine.Pin) *Device {
+	dcPin.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	resetPin.Configure(machine.PinConfig{Mode: machine.PinOutput})
+	csPin.Configure(machine.PinConfig{Mode: machine.PinOutput})
```

Comment:
Do we want to configure pins here? I think going forward we shouldn't configure pins in drivers...

Reply:
I'd like we re-define the contact so pins shall come initialized to the driver constructors.

But we don't want to break existing users and book and tutorials, I gather?

This is not a new driver, this PR is about eliminating heap leaks.